lstchain/image/muon/muon_analysis.py

```diff
@@ -87,19 +89,15 @@ def fit_muon(x, y, image, geom, tailcuts):
     image_clean = image * clean_mask
     muonringparam = muonring.fit(x, y, image_clean)
 
-    dist = np.sqrt(np.power(x - muonringparam.ring_center_x, 2)
-                       + np.power(y - muonringparam.ring_center_y, 2))
-    ring_dist = np.abs(dist - muonringparam.ring_radius)
-    muonringparam = muonring.fit(
-            x, y, image_clean * (ring_dist < muonringparam.ring_radius * 0.4)
-        )
-
-    dist = np.sqrt(np.power(x - muonringparam.ring_center_x, 2) +
-                       np.power(y - muonringparam.ring_center_y, 2))
-    ring_dist = np.abs(dist - muonringparam.ring_radius)
+    # Do an iterative fit removing pixels which are beyond 0.4*ring_radius of the ring (along the radial direction):
+    # (Q: is this effective? Seems anyway unnecessary for clean rings which are those we want in the end)
 
-    muonringparam = muonring.fit(
-            x, y, image_clean * (ring_dist < muonringparam.ring_radius * 0.4)
+    for _ in [0]*2:  # just to iterate the fit twice more
+        dist = np.sqrt(np.power(x - muonringparam.ring_center_x, 2)
+                       + np.power(y - muonringparam.ring_center_y, 2))
+        ring_dist = np.abs(dist - muonringparam.ring_radius)
+        muonringparam = muonring.fit(
+        x, y, image_clean * (ring_dist < muonringparam.ring_radius * 0.4)
```

Comment:
as we've been discussing, the 40% of the ring_radius may be a bit too much, specially in the case of using biased integrators. Maybe it is the time to change it?
One of your proposals was to relate it to the `ring_width`

Reply:
Also, in L.142 you define the `ring_integration_width`. If you change it later there, we will forget that here it was changed

Reply:
> as we've been discussing, the 40% of the ring_radius may be a bit too much, specially in the case of using biased integrators. Maybe it is the time to change it?

I would leave 40% here, because this is just to identify really far outliers.

> One of your proposals was to relate it to the `ring_width`

Now I am not sure that is the best for light integration, because then it will be somewhat dependent on the PSF. I would simply reduce it. I am setting it to 25% as default now.